interop/test-runner/main.go

```diff
@@ -530,22 +546,122 @@ func (config *ScriptActorConfig) ChangeCipherSuite() error {
 func (config *ScriptActorConfig) RunStep(index int, step ScriptStep) error {
 	switch step.Action {
 	case ActionCreateGroup:
-		client := config.ActorClients[step.Actor]
-		groupID := []byte(uuid.New().String())
-		req := &pb.CreateGroupRequest{
-			GroupId:          groupID,
-			CipherSuite:      config.CipherSuite,
-			EncryptHandshake: config.EncryptHandshake,
-			Identity:         []byte(step.Actor),
-		}
-		resp, err := client.rpc.CreateGroup(ctx(), req)
+		var params CreateGroupStepParams
+		err := json.Unmarshal(step.Raw, &params)
 		if err != nil {
 			return err
 		}
-		config.Log(step.Actor, "CreateGroup", req, resp)
 
-		config.stateID[step.Actor] = resp.StateId
-		config.StoreMessage(index, "group_id", groupID)
+		// Create the group
+		groupID := []byte(uuid.New().String())
+		{
+			client := config.ActorClients[step.Actor]
+			req := &pb.CreateGroupRequest{
+				GroupId:          groupID,
+				CipherSuite:      config.CipherSuite,
+				EncryptHandshake: config.EncryptHandshake,
+				Identity:         []byte(step.Actor),
+			}
+			resp, err := client.rpc.CreateGroup(ctx(), req)
+			if err != nil {
+				return err
+			}
+			config.Log(step.Actor, "CreateGroup", req, resp)
+
+			config.stateID[step.Actor] = resp.StateId
+			config.StoreMessage(index, "group_id", groupID)
+		}
+
+		// If there are now members to add, we're done
+		if len(params.Members) == 0 {
+			return nil
+		}
+
+		// Get key packages from the joiners
+		keyPackages := make([][]byte, len(params.Members))
+		transactionIDs := make([]uint32, len(params.Members))
+		for i, member := range params.Members {
+			client := config.ActorClients[member]
+			req := &pb.CreateKeyPackageRequest{
+				CipherSuite: config.CipherSuite,
+				Identity:    []byte(member),
+			}
+			resp, err := client.rpc.CreateKeyPackage(ctx(), req)
+			if err != nil {
+				return err
+			}
+			config.Log(member, "CreateKeyPackage", req, resp)
+
+			keyPackages[i] = resp.KeyPackage
+			transactionIDs[i] = resp.TransactionId
+		}
+
+		// Create and consume a Commit with inline Add proposals
+		var welcome []byte
+		var ratchetTree []byte
+		var epochAuthenticator []byte
+		{
+			byValue := make([]*pb.ProposalDescription, len(keyPackages))
+			for i, keyPackage := range keyPackages {
+				byValue[i] = &pb.ProposalDescription{
+					ProposalType: []byte("add"),
+					KeyPackage:   keyPackage,
+				}
+			}
+
+			client := config.ActorClients[step.Actor]
+			req := &pb.CommitRequest{
+				StateId: config.stateID[step.Actor],
+				ByValue: byValue,
+			}
+			resp, err := client.rpc.Commit(ctx(), req)
+			if err != nil {
+				return err
+			}
+			config.Log(step.Actor, "Commit", req, resp)
+
+			welcome = resp.Welcome
+			ratchetTree = resp.RatchetTree
```

Comment:
This way `req.ExternalTree = 0` so `resp.RatchetTree` should be empty. Maybe the `ratchetTree` variable can be removed?